java/arcs/core/data/HandleMode.kt

```diff
@@ -18,14 +18,26 @@ enum class HandleMode {
     Read,
     /** [Handle] is write only. */
     Write,
+    /** [Handle] is query only. */
+    Query,
     /** [Handle] is read-write. */
-    ReadWrite;
+    ReadWrite,
+    /** [Handle] is read-query. */
+    ReadQuery,
+    /** [Handle] is query-write. */
+    WriteQuery,
```

Comment:
Does a write have to match the query as well? Let's say I want a particle to be able to query and update data about California only in a schema that is Covid { state: Text, cases: Number, dead: Number }. Is it possible to block the particle writing entities with State == NY?

Reply:
No, read and write are separate behaviors to query.

A refinement of `state != NY` could be used to block the particle from writing data as you've suggested.